controller/volume_controller.go

```diff
@@ -435,6 +439,10 @@ func (vc *VolumeController) syncVolume(key string) (err error) {
 		}
 	}()
 
+	if err := vc.handleVolumeAttachmentCreation(volume); err != nil {
```

Comment:
IIUC, there are two methods to create a volume.
1. Create a volume and then create a volumeAttachement.
2. The legacy method is creating a volume and update volume.spec.nodeID.
Correct?

Reply:
When the volume is created the first time, this logic creates the volumeattachment CR so that we can add attachment tickets to it. I was thinking doing this one is webhook but cannot find a good way. If I create VA in webhook when the volume CR is created then we have this problem: we create the VA CR then volume CR. There would be a problem because the VA has the owner reference is the volume CR. But in this case the owner has not been created yet so Kubernetes might delete the newly created VA.

Therefore I move that VA creation into this volume controller. What do you think? @derekbit @shuo-wu @joshimoo?

Reply:
Yes. It's reasonable to create VAs in the controller rather than the webhook.